components/match2/wikis/stormgate/match_group_input_custom.lua

```diff
@@ -0,0 +1,807 @@
+---
+-- @Liquipedia
+-- wiki=stormgate
+-- page=Module:MatchGroup/Input/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Faction = require('Module:Faction')
+local Flags = require('Module:Flags')
+local HeroData = mw.loadData('Module:HeroData')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input', {requireDevIfEnabled = true})
+local Streams = Lua.import('Module:Links/Stream', {requireDevIfEnabled = true})
+
+local OpponentLibrary = require('Module:OpponentLibraries')
+local Opponent = OpponentLibrary.Opponent
+
+local ALLOWED_STATUSES = {'W', 'FF', 'DQ', 'L'}
+local CONVERT_STATUS_INPUT = {W = 'W', FF = 'FF', L = 'L', DQ = 'DQ', ['-'] = 'L'}
+local DEFAULT_LOSS_STATUSES = {'FF', 'L', 'DQ'}
+local MAX_NUM_OPPONENTS = 2
+local DEFAULT_BEST_OF = 99
+local LINKS_KEYS = {'preview', 'interview', 'review'}
+local MODE_MIXED = 'mixed'
```

Comment:
mixed would mean stuff like 1v2 or such (even 1vteam)
which is possible in custom games

Reply:
Rather leave stuff out until it's used, like with sideswipe. It may turn out to be needed one day, and then we add it. Otherwise it adds to maintenance without bringing any use

Reply:
well this is pretty much a 1 line addition
difference between having this string or nil as mode set in those cases
don't see the issue
like ned to do the check on opponentTypes anyways